### PR TITLE
Add azure virtual network ddos dashboard to cspmixin

### PR DESCRIPTION
**Pull request by rgeyer**
Adds an azure virtual network dashboard that will visualize the metrics available in that azure monitor namespace.

Unfortunately, those metrics are almost exclusively for DDoS mitigation stats. So we'll likely never see them. Thus, the included screenshot is.. Pretty barren.

It *should* work, in the case of an actual DDoS event...

image

### CODE REVIEW

File: csp-mixin/dashboards.libsonnet [hunk 1]
```diff
@@ -104,5 +104,22 @@ local commonlib = import 'common-lib/common/main.libsonnet';
                        csplib.grafana.rows.alb_loadbalancers
                      )
                    ),
+
+                 [csplib.config.uid + '-virtualnetwork.json']:
+                   local variables = csplib.signals.azureloadbalancer.getVariablesMultiChoice();
+                   g.dashboard.new(csplib.config.dashboardNamePrefix + 'Virtual Network')
```
Comment (v-zhuravlev) [inline csp-mixin/dashboards.libsonnet:110]: Sentence case?

File: csp-mixin/panels.libsonnet [hunk 2]
```diff
@@ -842,5 +842,71 @@ local commonlib = import 'common-lib/common/main.libsonnet';
...
+              index: 0,
+            },
+            '1': {
+              text: 'Under Attack',
+              color: 'dark-red',
+              index: 1,
+            },
+          },
+        }
+      ),
```
Comment (v-zhuravlev) [inline csp-mixin/panels.libsonnet:864]: btw, you can add mappings to signal itself, so it is added to any panel where this signal is attached.

File: csp-mixin/signals/azurevirtualnetwork.libsonnet [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        azuremonitor: {
+          expr: 'azure_microsoft_network_virtualnetworks_udpbytesdroppedddos_bytespersecond_maximum{%(queriesSelector)s}',
+          legendCustomTemplate: 'UDP Dropped',
+        },
+      },
+    },
+    
+    udpBytesForwarded: {
+      name: 'UDP bytes Forwarded DDoS',
+      description: 'Inbound udp bytes per second forwarded by DDoS protection.',
```
Comment (v-zhuravlev) [inline csp-mixin/signals/azurevirtualnetwork.libsonnet:106]:
```suggestion
      description: 'Inbound UDP bytes per second forwarded by DDoS protection.',
```

File: csp-mixin/signals/azurevirtualnetwork.libsonnet [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    },
+    
+    underDdos: {
+      name: 'Currently DDoS Status',
+      description: 'Is the system currently under DDoS attack or not.',
+      type: 'gauge',
+      sources: {
+        azuremonitor: {
+          expr: 'azure_microsoft_network_virtualnetworks_ifunderddosattack_count_maximum{%(queriesSelector)s}',
+          exprWrappers: [['','OR on() vector(0)']]
```
Comment (v-zhuravlev) [inline csp-mixin/signals/azurevirtualnetwork.libsonnet:303]: nice!

File: csp-mixin/rows.libsonnet [hunk 5]
```diff
@@ -216,5 +216,51 @@ local g = import './g.libsonnet';
...
+        + g.panel.timeSeries.gridPos.withH(6),
+
+        this.grafana.panels.vn_pingmesh_avg
+        + g.panel.timeSeries.gridPos.withW(6)
+        + g.panel.timeSeries.gridPos.withH(6),
+
+        this.grafana.panels.vn_packet_trigger
+        + g.panel.timeSeries.gridPos.withW(12)
+        + g.panel.timeSeries.gridPos.withH(6),
+      ],
```
Comment (v-zhuravlev) [inline csp-mixin/rows.libsonnet:234]: btw, you can now define all panels INSIDE row, like so:
https://github.com/grafana/jsonnet-libs/blob/master/kafka-observ-lib/rows.libsonnet#L6-L24

this was, you can decide if you want specific row to be collapsed or not collapsed and render less important rows as collapsed by by using g.util.panel.resolveCollapsedFlagOnRows() func:
https://github.com/grafana/jsonnet-libs/blob/master/kafka-observ-lib/dashboards.libsonnet#L17-L26

File: csp-mixin/signals/azurevirtualnetwork.libsonnet [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+      sources: {
+        azuremonitor: {
+          expr: 'azure_microsoft_network_virtualnetworks_udppacketsdroppedddos_countpersecond_maximum{%(queriesSelector)s}',
+          legendCustomTemplate: 'UDP Dropped',
+        },
+      },
+    },
+
+    udpPacketsForwarded: {
+      name: 'UDP packets Forwarded DDoS',
```
Comment (v-zhuravlev) [inline csp-mixin/signals/azurevirtualnetwork.libsonnet:215]: TItle or sentence case here?

Comment (rgeyer) [reply csp-mixin/signals/azurevirtualnetwork.libsonnet:215]: Yeah, good call. I'll clean these all up.. I'm the worst offender for this rule. Pretty sure I'm the reason it exists ;)

File: csp-mixin/panels.libsonnet (hunk 2 above)
Comment (rgeyer) [reply csp-mixin/panels.libsonnet:864]: Nice! I added this, thanks for pointing it out.

File: csp-mixin/rows.libsonnet (hunk 5 above)
Comment (rgeyer) [reply csp-mixin/rows.libsonnet:234]: Ahh nice! I don't really need any of these rows collapsed tho. Is there any other obvious reason I should consider the new format?

**Review by v-zhuravlev — APPROVED** [review]

**Pull request merged**